packages/vm/src/runTx.ts

```diff
@@ -207,6 +210,20 @@ export async function runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
 async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
   const state = this.stateManager
 
+  let stateAccesses
+  if (this.common.isActivatedEIP(6800)) {
+    if (!(this.stateManager instanceof StatelessVerkleStateManager)) {
+      throw Error(`StatelessVerkleStateManager needed for execution of verkle blocks`)
+    }
+    stateAccesses = (this.stateManager as StatelessVerkleStateManager).accessWitness
+  } else {
+    // assign an empty witness to prevent unnecessary code spagetti of null/undefined checks while usage
+    // but its totally possible that merkle can also accumulate witnesses if such feature was to be made
+    // available there
+    stateAccesses = new AccessWitness()
+  }
+  let txAccesses = stateAccesses?.shallowCopy()
```

Comment:
Currently this code block either points to the `this.stateManager` as VerkleStateManager and then shallow copy it (this is the same as a `new AccessWitness`) or it creates a new witness and then copies it (so does the same thing twice) :thinking:

Reply:
yes because this witness is just used to establish an upfront fee and then thrown away (rewrttiten by a new shallowCopy)